### ISSUE TITLE
[Wish] Automatic goal folding and windows size

### ISSUE THREAD

**Issue by gares** [coq-community/vscoq#872]
This is to track discussions on the goal folding/ellipses feature.

Ideally, the threshold should not be based on a fixed number but on the window size. If it cannot fit, then some parts should be folded. Making the window larger should unfold more. Making it smaller fold more. This also interacts with indentation, so it is not trivial.

**Comment by rtetley** [comment]
Good point ! I like the idea but it is indeed non trivial. How do we link folding depth to window size ?

**Comment by gares** [comment]
I have no idea, I don't know if it there is a known efficient algorithm.
A dumb one, a spec really, would be: if it does not fit, pick a leaf, elide it, goto 0.

Maybe a bisection? You elide half of the nodes, from the deepest ones, and see. If it fits you add half of what your removed, if it does not you remove that half (of half). And you iterate like 10 times max.

**Comment by rtetley** [comment]
Hmmm I'm thinking of changing the render algorithm anyways (w.r.t #859, #869) so maybe we can discuss it some time next week.
